ToneMapper/TMTonalFeature.m

```diff
@@ -33,19 +52,32 @@ @interface TMTonalFeature()
 /// \c UISegmentedControl object used to select filter operations to alter.
 @property (strong, nonatomic) UISegmentedControl *operatorOptions;
 
+/// \c UISegmentedControl object used to select advanced filter operations to alter.
+@property (strong, nonatomic) UISegmentedControl *advancedContrastOperatorOptions;
+
 @end
 
 @implementation TMTonalFeature
 
 @synthesize delegate = _delegate;
 
-- (instancetype)initWithDrawer:(TMGLDrawer *)drawer {
-  if (!drawer) {
+- (instancetype)initWithDrawer:(TMGLDrawer *)drawer andInputTexture:(TMGLTexture *)inputTexture {
+  if (!drawer || !inputTexture) {
     return nil;
   }
   
   if (self = [super init]) {
-    _filter = [[TMGLToneAdjustFilter alloc] initWithDrawer:drawer];
+      _inputTexture = inputTexture;
+      _bilateralFilter = [[TMGLBilateralFilter alloc] initWithDrawer:drawer];
+      self.bilateralFilter.kernelSize = kSmoothKernelSize;
+      TMGLTexture *smoothTexture = [self.bilateralFilter applyOnTexture:inputTexture];
+      self.bilateralFilter.kernelSize = kSmoothKernelSize;
```

Comment:
The problem is not the name, you use the same kernel size twice.

On Thu, Oct 29, 2015 at 1:44 PM, Barak Yoresh <notifications@github.com>
wrote:

> In ToneMapper/TMTonalFeature.m
> <https://github.com/barakyoresh/ToneMapper/pull/3#discussion_r43375112>:
>
> >      return nil;
> >    }
> >
> >    if (self = [super init]) {
> > -    _filter = [[TMGLToneAdjustFilter alloc] initWithDrawer:drawer];
> > +      _inputTexture = inputTexture;
> > +      _bilateralFilter = [[TMGLBilateralFilter alloc] initWithDrawer:drawer];
> > +      self.bilateralFilter.kernelSize = kSmoothKernelSize;
> > +      TMGLTexture *smoothTexture = [self.bilateralFilter applyOnTexture:inputTexture];
> > +      self.bilateralFilter.kernelSize = kSmoothKernelSize;
>
> Got a better name? :P
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/barakyoresh/ToneMapper/pull/3/files#r43375112>.
>